gemini/use-cases/retrieval-augmented-generation/utils/intro_multimodal_rag_utils.py

```diff
@@ -476,19 +440,17 @@ def get_image_metadata_df(
     """
 
     final_data_image: List[Dict] = []
-
     for key, values in image_metadata.items():
-        data: Dict = {}
-        data["file_name"] = filename
-        data["page_num"] = int(key) + 1
-
         for image_number, image_values in values.items():
+            data: Dict = {}
+            data["file_name"] = filename
+            data["page_num"] = key + 1
             data["img_num"] = int(image_values["img_num"])
             data["img_path"] = image_values["img_path"]
             data["img_desc"] = image_values["img_desc"]
-            data["mm_embedding_from_text_desc_and_img"] = image_values[
-                "mm_embedding_from_text_desc_and_img"
-            ]
+            # data["mm_embedding_from_text_desc_and_img"] = image_values[
+            #     "mm_embedding_from_text_desc_and_img"
+            # ]
```

Comment:
Why is this removed now?

Reply:
While going through the bug fixes, I realized that a) I am not using it anywhere, and b) it is different from what I thought it was. My understanding was that when you pass a text with images, the final embeddings are a combination, and both fuse to give one single embedding. 

However, having experimented with the new SDK, I see they are both different (and rightfully so; I expected too much ;) ). They are helpful when you have captions/descriptions for images with less than 32 tokens. In my case, that's not true. 

The combination might not work for typical enterprise documents (my assumption and empirical evidence), as captions associating images might be below or above the given images for different documents, and there is no easy way to resolve it. 

So, in my design of the logic, I keep the text (page)-text (Gemini image description) separate using text-embeddings from image (within the document)-image (within the document) using multimodal-embedding. 

Hence, when someone is trying to find an image using a text query, it will leverage text embeddings, and if they are trying to find an image using an image, then I use multimodal embeddings. They both can not exist in the same dimension space.

Reply:
Ok, I suppose that makes sense.